### DISCUSSION TITLE
TreeView vs TreeDataGrid

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#13933] [Q&A] [answered]
Hi,

for my project I am playing around, what is more useful, TreeView or TreeDataGrid.

In both cases I am running into limitations.

Is it possible for TreeView to show more than three levels?
Is it possible in TreeDataGrid to have child (and grand childs ...) nodes also expandable?

Using Treeview gives only nodes downto grand child, which would be (preliminary) enough but it is hard to build a TreeView with more columns as it TreeDataGrid does.

Using TreeDataGrid only provide the highest level as expandable.

Having TreeDataGrid with at least 3 expandable levels would be perfect.

Any chance?

3.97

**Comment by timunie** [comment]
Idea 3: Add a PR to TreeDataGrid to enable the feature you need
Idea 4: Use a ListBox with an MarginConverter and special styles. The bound colleciton needs to be filtered

**Reply by drei97** [reply, marked as answer]
I am sorry, I was wrong. DataTreeGrid is able to show more than one expandable level. My DataModel was not correct.
